src/lib/akeneo.ts

```diff
@@ -0,0 +1,194 @@
+import 'source-map-support/register';
+import axios, { AxiosResponse, AxiosError } from 'axios';
+import FormData from 'form-data';
+import { Akeneo } from 'akeneo';
+
+const GRANT_TYPE  = 'password';
+
+export interface AkeneoConfig {
+	host: string
+	username: string,
+	password: string,
+	oauthClientId: string,
+	oauthClientSecret: string,
+}
+
+interface PatchRequestData {
+	entity: string;
+	id: string;
+	// Because the Axios library takes an `any` as it's body type
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	body: any;
+	accessToken: string;
+}
+
+export class AkeneoClient {
+	config: AkeneoConfig;
+
+	constructor(config: AkeneoConfig) {
+		this.config = config
+	}
+
+	/**
+	 * Performs an Akeneo authentication request and returns the authentication details.
+	 * https://api.akeneo.com/documentation/authentication.html
+	 *
+	 * @return {Promise<AxiosResponse>} The response from the Akeneo API
+	 */
+	authenticate = (): Promise<Akeneo.AuthenticationResponse> => {
+		const { host, username, password, oauthClientId, oauthClientSecret } = this.config;
+
+		const data = {
+			username: username,
+			password: password,
+			grant_type: GRANT_TYPE
+		};
+
+		const tokenURL = `https://${host}/api/oauth/v1/token`;
+		const credentials = Buffer.from(`${oauthClientId}:${oauthClientSecret}`).toString('base64');
+
+		return axios.post(tokenURL, data, {
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': `Basic ${credentials}`
+			}
+		}).then(function (response) {
+			if (!("access_token" in response.data &&
+				"refresh_token" in response.data)) {
+				throw Error('Invalid repsonse from Akeneo / token endpoint.');
+			}
+
+			return {
```

Comment:
Future improvement note:
Might be a good idea to store the token and refresh token (and if there is a valid until datetime in the response) against the client so when it makes follow up requests it can just use what it already has and doesn't need to re-auth all the time.